In [2]:
import pandas as pd
from sqlalchemy import create_engine
import urllib.parse


In [3]:
password = urllib.parse.quote_plus("N@ndini2005")

engine = create_engine(
    f"postgresql+psycopg2://postgres:{password}@localhost:5432/ecommerce_db"
)

conn = engine.connect()
print("CONNECTED ✅")
conn.close()


CONNECTED ✅


In [7]:
df_final = pd.read_sql(
    "SELECT * FROM customer_behavior_final",
    engine
)

print("Data loaded ✅")
df_final.shape


Data loaded ✅


(1000000, 66)

In [8]:
segment_distribution = (
    df_final["customer_segment"]
    .value_counts(normalize=True) * 100
)

segment_distribution


customer_segment
general_segment    92.0162
target_segment      7.9838
Name: proportion, dtype: float64

In [9]:
df_final.groupby("customer_segment")["engagement_score"].mean()


customer_segment
general_segment    13.144588
target_segment     17.645094
Name: engagement_score, dtype: float64

In [10]:
df_final.groupby("customer_segment")["impulse_buying_score"].mean()


customer_segment
general_segment    4.853267
target_segment     6.671397
Name: impulse_buying_score, dtype: float64

In [11]:
df_final["recommended_category"].value_counts(normalize=True) * 100


recommended_category
home_appliances    54.5613
electronics        45.4387
Name: proportion, dtype: float64

In [14]:
segment_summary.to_sql(
    "final_business_insights",
    engine,
    if_exists="replace"
)

print("Business insights table saved ✅")

Business insights table saved ✅


In [13]:
segment_summary = df_final.groupby("customer_segment").agg({
    "engagement_score": "mean",
    "impulse_buying_score": "mean",
    "return_rate": "mean",
    "premium_subscription": "mean"
})

segment_summary



,engagement_score,impulse_buying_score,return_rate,premium_subscription
customer_segment,,,,
general_segment,13.144588,4.853267,50.011078,0.359515
target_segment,17.645094,6.671397,49.934304,0.358263


In [13]:
segment_summary = df_final.groupby("customer_segment").agg({
    "engagement_score": "mean",
    "impulse_buying_score": "mean",
    "return_rate": "mean",
    "premium_subscription": "mean"
})

segment_summary



,engagement_score,impulse_buying_score,return_rate,premium_subscription
customer_segment,,,,
general_segment,13.144588,4.853267,50.011078,0.359515
target_segment,17.645094,6.671397,49.934304,0.358263


In [13]:
segment_summary = df_final.groupby("customer_segment").agg({
    "engagement_score": "mean",
    "impulse_buying_score": "mean",
    "return_rate": "mean",
    "premium_subscription": "mean"
})

segment_summary



,engagement_score,impulse_buying_score,return_rate,premium_subscription
customer_segment,,,,
general_segment,13.144588,4.853267,50.011078,0.359515
target_segment,17.645094,6.671397,49.934304,0.358263


In [13]:
segment_summary = df_final.groupby("customer_segment").agg({
    "engagement_score": "mean",
    "impulse_buying_score": "mean",
    "return_rate": "mean",
    "premium_subscription": "mean"
})

segment_summary



,engagement_score,impulse_buying_score,return_rate,premium_subscription
customer_segment,,,,
general_segment,13.144588,4.853267,50.011078,0.359515
target_segment,17.645094,6.671397,49.934304,0.358263


In [13]:
segment_summary = df_final.groupby("customer_segment").agg({
    "engagement_score": "mean",
    "impulse_buying_score": "mean",
    "return_rate": "mean",
    "premium_subscription": "mean"
})

segment_summary



,engagement_score,impulse_buying_score,return_rate,premium_subscription
customer_segment,,,,
general_segment,13.144588,4.853267,50.011078,0.359515
target_segment,17.645094,6.671397,49.934304,0.358263


In [13]:
segment_summary = df_final.groupby("customer_segment").agg({
    "engagement_score": "mean",
    "impulse_buying_score": "mean",
    "return_rate": "mean",
    "premium_subscription": "mean"
})

segment_summary



,engagement_score,impulse_buying_score,return_rate,premium_subscription
customer_segment,,,,
general_segment,13.144588,4.853267,50.011078,0.359515
target_segment,17.645094,6.671397,49.934304,0.358263


segment_summary = df_final.groupby("customer_segment").agg({
    "engagement_score": "mean",
    "impulse_buying_score": "mean",
    "return_rate": "mean",
    "premium_subscription": "mean"
})

segment_summary


# 📊 Step 4: Business Insights Analysis

## 🔎 Segment Comparison

### 1️⃣ Engagement Score
- Target Segment: 17.64
- General Segment: 13.14  
✅ Target customers are more engaged.

### 2️⃣ Impulse Buying Score
- Target Segment: 6.67
- General Segment: 4.85  
✅ Target customers show higher impulse buying behavior.

### 3️⃣ Return Rate
- Both segments have similar return rates (~50%)  
⚠ No major difference in returns.

### 4️⃣ Premium Subscription Rate
- Both segments around 35%  
📌 Subscription behavior is similar.

---

## 🎯 Business Recommendation

- Promote **electronics** to Target Segment.
- Run personalized marketing campaigns.
- Increase premium subscription benefits for high engagement users.
- Monitor return rates to reduce product returns.

---

## 📈 Conclusion

Target Segment customers are:
✔ More engaged  
✔ Higher impulse buyers  
✔ Good candidates for premium product promotions  

This segment should be the primary focus for revenue growth
